src/bot.coffee

```diff
@@ -39,6 +41,7 @@ class SlackBot extends Adapter
     @client.rtm.on "close", @close
     @client.rtm.on "disconnect", @disconnect
     @client.rtm.on "error", @error
+    @client.web.on Events.WEB.RATE_LIMITED, @rate_limited
```

Comment:
@mistydemeo Here is a sample I used for verifying this implementation.

```coffeescript
module.exports = (robot, options) ->
  web = robot.adapter.client.web
  
  robot.hear /call a lot/i, (res) ->
    for x in [1..100]
      web.conversations.list().then (res) ->
        robot.logger.info "#{res.ok} #{res.error}"

  robot.on 'slack-rate-limit', (res) ->
    robot.logger.info "Detected a rate-limited call #{res}"
```


Does this really work for you? It doesn't work for me. The following is the error I get every time the `rate_limited` method is called. I haven't investigated details yet but it seems `@robot` is undefined in any case.

```
[Fri Apr 10 2020 13:05:29 GMT+0900 (Japan Standard Time)] ERROR TypeError: Cannot read property 'emit' of undefined
  at WebAPIClient.SlackBot.rate_limited (/path-to-project/node_modules/hubot-slack/src/bot.coffee:217:5, <js>:276:25)
  at WebAPIClient.emit (/path-to-project/node_modules/hubot-slack/node_modules/eventemitter3/index.js:116:35)
  at handleRateLimitResponse (/path-to-project/node_modules/hubot-slack/node_modules/@slack/client/lib/clients/transports/call-transport.js:51:10)
  at handleTransportResponse (/path-to-project/node_modules/hubot-slack/node_modules/@slack/client/lib/clients/transports/call-transport.js:148:21)
```

Also, to utilize this, you need to always use `robot.adapter.client.web`. If some of your scripts initialize `WebClient` separately (also, that's the way this adapter currently recommends), rate-limited errors those clients get won't be notified here.

Correct me if my understanding is wrong.